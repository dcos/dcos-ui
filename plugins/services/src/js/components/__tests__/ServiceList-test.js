jest.dontMock('../../structs/ServiceTree');
jest.dontMock('../ServiceList');
jest.dontMock('../../stores/MarathonStore');
/* eslint-disable no-unused-vars */
const React = require('react');
/* eslint-enable no-unused-vars */
const ReactDOM = require('react-dom');

const ServiceList = require('../ServiceList');
const ServiceTree = require('../../structs/ServiceTree');

describe('ServiceList', function () {

  describe('#shouldComponentUpdate', function () {

    beforeEach(function () {
      var services = new ServiceTree({items: [{name: 'foo'}]});
      this.container = document.createElement('div');
      this.instance = ReactDOM.render(
        <ServiceList
          services={services.getServices().getItems()}
          healthProcessed={false} />,
        this.container
      );
    });

    afterEach(function () {
      ReactDOM.unmountComponentAtNode(this.container);
    });

    it('should allow update', function () {
      var shouldUpdate = this.instance.shouldComponentUpdate({a: 1});
      expect(shouldUpdate).toEqual(true);
    });

    it('should not allow update', function () {
      var shouldUpdate = this.instance.shouldComponentUpdate(
        this.instance.props
      );
      expect(shouldUpdate).toEqual(false);
    });

  });

  describe('#getServices', function () {

    beforeEach(function () {
      var services = new ServiceTree({items: [{name: 'foo'}]});
      this.container = document.createElement('div');
      this.instance = ReactDOM.render(
        <ServiceList
          services={services.getServices().getItems()}
          healthProcessed={false} />,
        this.container
      );
    });

    afterEach(function () {
      ReactDOM.unmountComponentAtNode(this.container);
    });

    it('returns services that have a value of two elements', function () {
      var services = new ServiceTree({items: [{name: 'foo'}]})
        .getServices().getItems();
      var result = this.instance.getServices(services, false);

      expect(result[0].content[0].content.key).toEqual('title');
      expect(result[0].content[1].content.key).toEqual('health');
    });
  });

});
