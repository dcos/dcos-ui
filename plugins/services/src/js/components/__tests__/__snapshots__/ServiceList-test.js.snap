// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceList #getServices returns services that have a value of two elements 1`] = `
<div
  className="dashboard-health-list"
>
  <ul
    className="list list-unstyled"
  >
    <li
      className="list-item"
    >
      <span
        className="dashboard-health-list-item-description text-overflow"
      >
        <a
          className="dashboard-health-list-item-cell emphasis"
          href={null}
          onClick={[Function]}
        >
          
        </a>
      </span>
      <div
        className="dashboard-health-list-health-label"
      >
        <div
          className="tooltip-wrapper"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <svg
            aria-label="system-circle-minus icon"
            className="css-ed86y8"
            height={16}
            preserveAspectRatio="xMinYMin meet"
            role="img"
            viewBox="0 0 16 16"
            width={16}
          >
            <use
              xlinkHref="#system-circle-minus"
            />
          </svg>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
